CMakeLists.txt

```diff
@@ -186,6 +186,15 @@ install(
     RUNTIME DESTINATION bin
 )
 
+# Testing
+option( E57_BUILD_TEST "Build tests" ${E57_BUILDING_SELF} )
+
+if ( E57_BUILD_TEST )
+    enable_testing()
+
+    add_subdirectory( test )
+endif()
+
 # CMake package files
```

Comment:
Perhaps some preference _not_ to auto-enable testing.  (opt-in?)
We have some very large mono-repo style builds that would likely want to skip that step.

Reply:
On the other hand I am sympathetic to the "test by default" school of thinking.

Reply:
I was going for an "in-between" solution. If you are building libE57Format directly it should be enabled by default (`${E57_BUILDING_SELF}`). If it's included as a subdirectory, it should be disabled by default.

Reply:
Yes, entirely reasonable.